### PR TITLE
bug 1079920

### PR DESCRIPTION
**Pull request by trevorhobson**
Make parsing Feature header in compatibility table more robust
Add IE to Internet Explorer name fix
Add handling of additional Compat templates

### CODE REVIEW

**Comment by jwhitlock** [comment]
Thank you for your contribution.  It looks like a few changes are needed before I can merge:

- ``PageVisitor.visit_compat_headers`` needs its index for the cells list updated
- A test is needed for ``CompatAndroid`` for test coverage
- The line needs to be split up for pep8 conformance.  Maybe something like:converted to a list, and something like
```python
kuma_compat_versions = [
    'compat' + b for b in ('android', 'chrome', 'ie', ...)]
...
elif kname in kuma_compat_versions:
```

Please ensure `make qa` is happy with your pull request.

**Comment by trevorhobson** [comment]
Sorry I have no idea what is required to be changed for "PageVisitor.visit_compat_headers needs its index for the cells list updated"

The other 2 points no problem I expect.

The make stuff is beyond me.

File: mdn/scrape.py [hunk 1]
```diff
@@ -90,7 +90,7 @@
 compat_div_id = qtext
 compat_table = "<table class=\"compat-table\">" _ compat_body _ "</table>" _
 compat_body = "<tbody>" _ compat_headers _ compat_rows* _ "</tbody>"
-compat_headers = tr_open _ "<th>Feature</th>" _ compat_client_cells _ "</tr>" _
+compat_headers = tr_open _ th_open _ "Feature</th>" _ compat_client_cells _ "</tr>" _
```
Comment (jwhitlock) [inline mdn/scrape.py]: ``children[0]`` is tr_open, ``children[1]`` is _ (whitespace).  ``children[2]`` used to be ``<th>Feature</th>``, but now it is th_open.  Adding that moved the index for ``compat_client_cells``, which is why the assertion is failing.

**Comment by jwhitlock** [comment]
There is an assertion that is causing scraping and the tests to fail, because of the change to the `compat_headers` rule:

https://github.com/jwhitlock/web-platform-compat/pull/22/files#diff-ad77ec6aef81e1c517c65aae19c0f44dL491

You can run tests with `make test` or `./manage.py test mdn`.   You can also see the failing tests on Travis CI:

https://travis-ci.org/jwhitlock/web-platform-compat/jobs/50485455

Are you having problems running `make qa`, or working with the results?

**Comment by trevorhobson** [comment]
Thanks for the info on the rule.

I assume that to run the tests with make you have to do the https://web-platform-compat.readthedocs.org/en/latest/installation.html first? I get stuck at step 2. If this is the case then it would be useful to me to have a list of actual commands required to complete the steps as I know nothing of virtualenv, pip etc. If it is not the case then I am just having problems running `make qa`.

I can see failing tests on Travis CI, but they seemed to me to be happening after anything related to what I had changed, so I could not figure them out.

**Comment by jwhitlock** [comment]
I'll update the docs to add links to the tools, and a sample output from running on OS X.  However, installing these tools is very OS dependent, so I'll have to refer you to the upstream docs:

* http://pip.readthedocs.org/en/stable/ - install Python packages
* https://virtualenv.pypa.io/en/latest/index.html - Create "virtual environments" to allow multiple Python projects on the same computer 
* http://virtualenvwrapper.readthedocs.org/en/latest/ - Additional scripts to make `virtualenv` easier to use.

You'll probably also need the developer tools, such as Xcode for OS X or build-essential on Ubuntu.

Here's sample commands for a fresh project:
```sh
$ cd ~/src
$ git clone git@github.com:jwhitlock/web-platform-compat.git
Cloning into 'web-platform-compat'...
...
Checking connectivity... done.
$ cd web-platform-compat
$ mkvirtualenv wpc
New python executable in wpc/bin/python2.7
Also creating executable in wpc/bin/python
Installing setuptools, pip...done.
(wpc)$ pip install -r requirements.txt -r requirements.dev.txt
Downloading/unpacking https://github.com/jwhitlock/drf-json-api/archive/v0.1d.zip (from -r requirements.txt (line 21))
...
Successfully installed Django MarkupSafe Jinja2 jingo djangorestframework Markdown django-filter six django-extensions django-simple-history dj-database-url nose django-nose mock gunicorn psycopg2 static3 dj-static pytz pylibmc django-pylibmc django-pylibmc-sasl billiard amqp anyjson kombu celery django-mptt django-sortedm2m drf-cached-instances requests django-cors-headers Pygments Sphinx docutils py virtualenv tox mccabe pep8 pyflakes flake8 pep257 flake8-docstrings check-manifest pyroma Werkzeug gnureadline ipython ipdb sqlparse django-debug-toolbar coverage docopt PyYAML coveralls drf-json-api
Cleaning up...
(wpc)$ export DJANGO_DEBUG=1
(wpc)$ ./manage.py syncdb
Operations to perform:
  Synchronize unmigrated apps: sortedm2m, mptt, corsheaders, simple_history, debug_toolbar, django_extensions, rest_framework
...
You have installed Django's auth system, and don't have any superusers defined.
Would you like to create one now? (yes/no): yes
Username (leave blank to use 'john'): john
Email address: john@factorialfive.com
Password: 
Password (again): 
Superuser created successfully.
(wpc)$  ./manage.py runserver
Performing system checks...

System check identified no issues (0 silenced).
February 13, 2015 - 21:12:01
Django version 1.7.1, using settings 'wpcsite.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
...
(CTRL-C to cancel)
(wpc)$ make qa
flake8 .
coverage erase
coverage run --source webplatformcompat,tools setup.py test
running test
running egg_info
...
webplatformcompat/static/js/browse.js is OK.

webplatformcompat/static/js/webplatformcompat.js is OK.
(wpc)$ make test # to just run tests
```

**Comment by trevorhobson** [comment]
I got it working eventually, thanks for giving me enough details to get me there. On my Mint 17 I had to do the following:
sudo apt-get install libpq-dev lib32ncurses5-dev libmemcached-dev python-dev python-pip
sudo pip install setuptools virtualenv virtualenvwrapper --upgrade

Had to do the upgrade because I first tried to install the packages through apt and they did not work.

**Pull request merged**